lock-api/src/main/java/com/palantir/lock/client/timestampleases/MinLeasedTimestampGetter.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.client.timestampleases;
+
+import com.palantir.atlasdb.timelock.api.TimestampLeaseName;
+import java.util.Map;
+import java.util.Set;
+
+public interface MinLeasedTimestampGetter extends AutoCloseable {
```

Comment:
I wonder: Do we need to autobatch this for now? I'd say autobatching the acquire leased timestamp is much more important, because that will be coming from multiple threads.

This is currently internally a single task that will be doing this, we can punt autobatching this.

Reply:
It's probably worth doing if only because internal product has lots of clients/namespaces?

Reply:
Yes, it's just a question of ordering: we can start testing on single-atlasdb products, without waiting on this.

Reply:
And even then, we don't have this right now and things are fine. We won't introduce any new traffic yet, so it's 1:1 endpoint change, what's the big deal?

Reply:
Discussed offline: there's not much more work from here, because it's the simple autobatchign case. So let's just merge this